fulp_modules/features/antagonists/bloodsuckers/code/powers/targeted/_powers_targeted.dm

```diff
@@ -62,39 +62,31 @@
 	return istype(target_atom)
 
 /// Click Target
-/datum/action/bloodsucker/targeted/proc/ClickWithPower(atom/target_atom)
+/datum/action/cooldown/bloodsucker/targeted/proc/click_with_power(atom/target_atom)
 	// CANCEL RANGED TARGET check
 	if(power_in_use || !CheckValidTarget(target_atom))
 		return FALSE
 	// Valid? (return true means DON'T cancel power!)
-	if(!CheckCanPayCost() || !CheckCanUse(owner) || !CheckCanTarget(target_atom))
+	if(!can_pay_cost() || !can_use(owner) || !CheckCanTarget(target_atom))
 		return TRUE
 	power_in_use = TRUE // Lock us into this ability until it successfully fires off. Otherwise, we pay the blood even if we fail.
 	FireTargetedPower(target_atom) // We use this instead of ActivatePower(trigger_flags), which has no input
 	// Skip this part so we can return TRUE right away.
 	if(power_activates_immediately)
-		PowerActivatedSuccessfully() // Mesmerize pays only after success.
+		power_activated_sucessfully() // Mesmerize pays only after success.
 	power_in_use = FALSE
 	return TRUE
 
 /// Like ActivatePower, but specific to Targeted (and takes an atom input). We don't use ActivatePower for targeted.
-/datum/action/bloodsucker/targeted/proc/FireTargetedPower(atom/target_atom)
+/datum/action/cooldown/bloodsucker/targeted/proc/FireTargetedPower(atom/target_atom)
 	log_combat(owner, target_atom, "used [name] on")
 
 /// The power went off! We now pay the cost of the power.
-/datum/action/bloodsucker/targeted/proc/PowerActivatedSuccessfully()
+/datum/action/cooldown/bloodsucker/targeted/proc/power_activated_sucessfully()
 	unset_click_ability(owner)
-	PayCost()
-	StartCooldown()	// Do AFTER UpdateIcon() inside of DeactivatePower. Otherwise icon just gets wiped.
+	pay_cost()
+	StartCooldown()
 	DeactivatePower()
 
-/datum/action/bloodsucker/targeted/proc/InterceptClickOn(mob/living/caller, params, atom/target)
-	ClickWithPower(target)
-
-/datum/action/bloodsucker/proc/set_click_ability(mob/on_who)
```

Comment:
Is this done on the cooldown parent?

Reply:
set_click is i'm pretty sure + InterceptClickOn can't be a proc because it's already defined on parent

Reply:
yeah I meant set_click_ability and unset